xio-core/src/main/java/com/xjeffrose/xio/firewall/ConnectionLimiter.java

```diff
@@ -1,37 +1,49 @@
-package com.xjeffrose.xio.server;
+package com.xjeffrose.xio.firewall;
 
+import static com.codahale.metrics.MetricRegistry.name;
+
+import com.codahale.metrics.Counter;
+import com.codahale.metrics.MetricRegistry;
+import com.xjeffrose.xio.server.XioServerLimits;
 import io.netty.channel.ChannelDuplexHandler;
+import io.netty.channel.ChannelFutureListener;
 import io.netty.channel.ChannelHandler;
 import io.netty.channel.ChannelHandlerContext;
 import java.util.concurrent.atomic.AtomicInteger;
 import lombok.extern.slf4j.Slf4j;
 
-@ChannelHandler.Sharable
 @Slf4j
-public class XioConnectionLimiter extends ChannelDuplexHandler {
-
+@ChannelHandler.Sharable
+public class ConnectionLimiter extends ChannelDuplexHandler {
   private final AtomicInteger numConnections;
   private final int maxConnections;
+  private final Counter connections;
 
-  public XioConnectionLimiter(int maxConnections) {
-    this.maxConnections = maxConnections;
+  public ConnectionLimiter(MetricRegistry metrics, XioServerLimits limits) {
+    this.maxConnections = Math.min(limits.maxConnections(), 15000);
     this.numConnections = new AtomicInteger(0);
+    this.connections = metrics.counter(name("Active Connections"));
   }
 
   @Override
   public void channelActive(ChannelHandlerContext ctx) throws Exception {
+    connections.inc();
+
+    // TODO(JR): Should this return a 429 or is the current logic of silently dropping the
+    // connection sufficient?
     if (maxConnections > 0) {
       if (numConnections.incrementAndGet() > maxConnections) {
-        ctx.channel().close();
-        // numConnections will be decremented in channelClosed
-        log.info("Accepted connection above limit (" + maxConnections + "). Dropping.");
+        log.info("Accepted connection above limit (%d). Dropping.", maxConnections);
```

Comment:
`%d` should be `{}`

https://static.javadoc.io/org.slf4j/slf4j-api/1.7.25/org/slf4j/Logger.html